src/ngMessages/messages.js

```diff
@@ -719,6 +715,10 @@ function ngMessageDirectiveFactory() {
             }
           }
         });
+
+        scope.$on('$destroy', function() {
```

Comment:
There should be a comment here why this extra listener is necessary. 
Btw, is this the new scope that belongs only to the message? Or the parent scope?

Reply:
I'll add a comment explaining this.

This directive (and the `ngMessages` directive) do not create new scopes. So the scope we are listening to is the one that contains this `ngMessage` directive. E.g. the scope created by an `ngIf` or `ngRepeat` directive that wraps this `ngMessage`.